src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -2,28 +2,42 @@
 
 import com.es.phoneshop.dao.ArrayListProductDao;
 import com.es.phoneshop.dao.ProductDao;
+import com.es.phoneshop.exception.OutOfStockException;
+import com.es.phoneshop.model.cart.Cart;
 import com.es.phoneshop.model.product.*;
+import com.es.phoneshop.service.CartService;
+import com.es.phoneshop.service.HttpSessionCartService;
 import com.es.phoneshop.service.HttpSessionRecentlyViewedService;
 import com.es.phoneshop.service.RecentlyViewedService;
+import org.apache.http.client.utils.URIBuilder;
 
 import javax.servlet.ServletConfig;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 
 public class ProductListPageServlet extends HttpServlet {
 
     private ProductDao productDao;
     private RecentlyViewedService recentlyViewedService;
+    private CartService cartService;
+    private Map<String, String> params = new HashMap<>();
```

Comment:
A servlet instance is used for all requests from all users. You cannot have a common state of request parameters here. Besides, making stateful classes is always considered a bad practice.